### PR TITLE
Hello 👋🏻

### PR DESCRIPTION
**Pull request by WaynePatrickDalton**
Been a whirlwind few weeks, so good to get back working on Binoculars. A couple of things happening this weekend in my branch "structure":

1: I'm testing out Rider IDE (I have a free account for a year, so why not!).
2. Went for a restructure first, as an option for the team to look at, and partly to satisfy my OCD.
3. I'm looking at WPF for a simple SplashScreen for fun. Turning out to be quite painful, Rider is not fantastic at this, either am I for that matter. Mush Less fun than anticipated.
4. Creating space for Excel, Amazon, CSV, JSON, and Google Sheets options.
5. Brought back the Extension and Extension definition that was removed - everything was okay until then - now app not featuring on the Dynamo UI.
6. This Branch is still building, and copying to the Dynamo Core\2.0\packages\Binoculars folder. But no joy. Will investigate.
7. Popped in a few more Todo's etc.
8. Comments galore to help explain what's happening for newbies.

Few more ideas to add on structure, and framing out features etc. Feel free to jump in and help out, as my C# is not terribly fantastic 😣

Wayne 😊

### CODE REVIEW

**Comment by WaynePatrickDalton** [comment]
Morning @radumg & @brenncass,

Just a quick question (probably a rookie error!): 

**What would cause Binoculars not to load onto the UI in Dynamo?** 

- I did some chopping and changing in the files - and everything was going completely fine. 
-Then I brought back our old Extension and Extension Definition file to look at #27 -  and have broken something.
- Everything seems correct to my eyes in the .csproj actions. 
- All seems to delete and copy to the Sandbox Packages Path.
- Compiling in both Rider and Visual Studio just fine, but not loading onto the UI 🤦‍♂️

If you have any ideas let me know 👍🏻

**Comment by radumg** [comment]
You mention making changes to files. Have you updated the *ExtensionDefinition.XML file ? Either it’s pointing to wrong path to load DLL (file renamed or moved) or the class name / namespace is now wrong.

Typed on a tiny screen

**Comment by WaynePatrickDalton** [comment]
Went back and retracted myself in the Master, it happens when I move the definitions into a separate folder. So, the assembly path in the _ExtensionDefinition.xml_ is incorrect most likely. 

I'll have a look at your _Hello Dynamo_ examples again @radumg, and see if I can get my head around.

**Comment by brencass** [comment]
As Radu has said double check everything is fine, eg check namespaces/references are all still tied together correctly after any renaming or changing(double check casing/spelling). I noticed you had added "Extension" to the end of a name space, make sure this sort of thing is fully followed through to everything.

Last possible thing for you to check is to open a brand new dynamo file after dynamo is loaded and look into the log.

**To get to the logger**: hover your mouse over the bottom of dynamo at the red line location in the below image then drag it up after the mouse icon has changed to a double black arrow.

image

Once you are here you can scroll up and down and see if there is any related errors, as these might guide you towards a issue.

**Comment by brencass** [comment]
This is the error i get in dynamo
Capture

In your project you may want to create a json files for your configs then have the relevant data blank, and do add a checker to maybe not load a part of the addon if the config settings are blank?

I had to create a blank version of these files without any knowledge of there required structure/input requirements, this could be why i was getting my error and not related to your error?

**Comment by radumg** [comment]
> This is the error i get in dynamo
> Capture

yes, this is a path error, Dynamo can't find your extension `.dll`

**Comment by radumg** [comment]
> 1: I'm testing out Rider IDE (I have a free account for a year, so why not!).
> 3. I'm looking at WPF for a simple SplashScreen for fun. Turning out to be quite painful, Rider is not fantastic at this, either am I for that matter. Mush Less fun than anticipated.

ditch it and go for VS2019 😎, just making your life harder than it needs to be.

File: Definitions/Binoculars_ExtensionDefinition.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<ExtensionDefinition>
+  <AssemblyPaths>..\bin\Binoculars.dll</AssemblyPaths>
+  <TypeName>Binoculars.Extension.Extension</TypeName>
```
Comment (radumg) [inline Definitions/Binoculars_ExtensionDefinition.xml]: doubled-up namespaces are not fun 🤓

**Comment by StudioLE** [comment]
@WaynePatrickDalton Thisssssssssssss ssssssssssssshould fixsssssssssssssss it for you: https://github.com/teamtreedyn/Binoculars/pull/28/commits/9f59e0c14d39ec36bbbe0e131ed2343a0125c283

**Pull request closed without merging**